### ISSUE TITLE
docs: need a bit more clarity on the pipeline?

### ISSUE THREAD

**Issue by lrlna** [pull-stream/pull-stream#83]
hiya! 👋🏾

i was trying to implement a lil' pipeline using `pull-stream` and found it a little difficult (took a solid hour i'm p sure) to figure out what exactly / how exactly to get it to work with `pull-stream` itself. 

i wanted to give you a more concrete suggestion than just a 😭-i-can't-understand-your-docs type of thing, and i think i pinned the issue down to the fact that i didn't find out that i look at a single pipeline when creating a `pull-stream`, i.e. 
```javascript
pull(source, through, sink) 
```
until much later, since the original example just does a pull on some files
```javascript
pull.values(['file1', 'file2', 'file3'])
```
so basically you end up reading through what a source, through and sink are, then going through [examples](https://github.com/dominictarr/pull-stream-examples/blob/master/pull.js), and only then figuring out what you do with all those examples. which makes me think that i'd probably just like to see this [pipeline thing](https://github.com/pull-stream/pull-stream#pipeability) a lot earlier, since every streams package does pipeability differently and i would like to know how exactly yours is different / better.

idk if that makes sense? let me know if it doesn't and i can try explaining a bit more 

thanks folks! ^_^

**Comment by dominictarr** [comment]
This is good feedback, thank you
